www/grpc/network.go

```diff
@@ -106,23 +96,47 @@ func (s *networkServer) GetNetworkInfo(_ context.Context,
 		return false
 	})
 
-	sentBytes := make(map[int32]int64)
-	for msgType, bytes := range ps.SentBytes() {
-		sentBytes[int32(msgType)] = bytes
-	}
-
-	receivedBytes := make(map[int32]int64)
-	for msgType, bytes := range ps.ReceivedBytes() {
-		receivedBytes[int32(msgType)] = bytes
-	}
-
 	return &pactus.GetNetworkInfoResponse{
-		TotalSentBytes:      ps.TotalSentBytes(),
-		TotalReceivedBytes:  ps.TotalReceivedBytes(),
 		NetworkName:         s.net.Name(),
 		ConnectedPeersCount: uint32(len(peerInfos)),
 		ConnectedPeers:      peerInfos,
-		SentBytes:           sentBytes,
-		ReceivedBytes:       receivedBytes,
+		MetricInfo:          metricToProto(ps.Metric()),
 	}, nil
 }
+
+func metricToProto(m metric.Metric) *pactus.MetricInfo {
```

Comment:
I think instead return Metric object, use Get method + locker or `sync.Map` for read metric details.

Reply:
Here read fields of `Metric` may data race happen.

Reply:
Yes here is a copy of the metric object.